python/experiments/generate-box-tests-reports.main.kts

```diff
@@ -285,8 +285,12 @@ fun generateGitHistoryPlot(gitHistoryPlotPath: Path) {
     ggsave(p, gitHistoryPlotPath.fileName.toString(), path = gitHistoryPlotPath.parent?.toString() ?: ".")
 }
 
-fun Repository.getNumberOfFailedTests(tree: RevTree) =
-    readFileAsText(tree, Paths.get("python/experiments/failed-tests.txt"))?.lines()?.size
+fun Repository.getNumberOfFailedTests(tree: RevTree): Int? {
+    // Trying two paths, to be able to fetch number of failed tests for various file layouts that the repository had.
+    val failedTestsFile = readFileAsText(tree, Paths.get("python/experiments/failed-tests.txt"))
+        ?: readFileAsText(tree, Paths.get("python/box.tests/reports/pythonTest/failed-tests.txt"))
```

Comment:
Duplicated code 😉 We can use collections here like the following (I haven't tested this, maybe adjustments are needed, please test if you decide to apply this change):

```suggestion
    val failedTestsFile = listOf(
        "python/experiments/failed-tests.txt",
        "python/box.tests/reports/pythonTest/failed-tests.txt",
    ).firstNotNullOfOrNull { readFileAsText(tree, Paths.get(it)) }
```

Reply:
Thanks, however I expect this code to undergo other refactorings in the near future, so let me hold with such cleanup.